GDJS/Runtime/spriteruntimeobject.js

```diff
@@ -4,6 +4,62 @@
  * This project is released under the MIT License.
  */
 
+// Type Definitions
+
+/**
+ * @typedef {Object} Point Represents a Point in a frame
+ * @property {number} x X position of the point
+ * @property {number} y Y position of the point
+ */
+
+/**
+ * @typedef {Object} CustomPointData Represents a Custom Point in a frame
+ * @property {string} name Name of the point (To access it later)
+ * @property {number} x X position of the point
+ * @property {number} y Y position of the point
+ */
+
+/**
+ * @typedef {Object} CenterPointData Represents the center Point in a frame
+ * @property {boolean} automatic Is Automatically found or predefined?
+ * @property {number} x X position of the point
+ * @property {number} y Y position of the point
+ */
+
+/**
+ * @typedef {Object} FrameData Represents a {@link gdjs.SpriteAnimationFrame}
+ * @property {string} [image] The resource name of the image used in this frame
+ * @property {Array<customPointData>} [points] The points of the frame
+ * @property {point} originPoint The origin point
+ * @property {centerPointData} centerPoint The center of the frame
+ * @property {boolean} hasCustomCollisionMask Is The collision mask custom?
+ * @property {Array<Array<point>>} [customCollisionMask] The collision mask if it is custom
+ */
+
+/**
+ * @typedef {Object} DirectionData Represents the data of a {@link gdjs.SpriteAnimationDirection}
+ * @property {number} timeBetweenFrames Time between each frame (FPS)
+ * @property {boolean} looping Is the animation looping?
+ * @property {Array<gdjs.SpriteAnimationFrame>} sprites The list of frames
+ */
+
+/**
+ * @typedef {Object} AnimationData Represents the data of a {@link gdjs.SpriteAnimation}
+ * @property {string} [name] The name of the animation
+ * @property {boolean} useMultipleDirections Does the animation use multiple {@link gdjs.SpriteAnimationDirection}?
+ * @property {Array<DirectionData>} directions The list of {@link DirectionData} representing {@link gdjs.SpriteAnimationDirection} instances
+ */
+
+/**
+ * @typedef {Object} SpriteObjectDataType
+ * @property {boolean} updateIfNotVisible Update the object even if he is not visible?
+ * @property {Array<AnimationData>} animations The list of {@link AnimationData} representing {@link gdjs.SpriteAnimation} instances
+ * 
+ * @typedef {ObjectData & SpriteObjectDataType} SpriteObjectData
+ */
+
+// Code
```

Comment:
No need to specify "// Code", it's a bit redundant as everything is code.
"// Object implementation" can work, but I think it's useless comments, `typedef` is already specifying that you're writing a type definition, and the code is self describing.

Reply:
Fixed